test/widgets/subscription_list_test.dart

```diff
@@ -0,0 +1,146 @@
+import 'package:checks/checks.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:zulip/api/model/initial_snapshot.dart';
+import 'package:zulip/api/model/model.dart';
+import 'package:zulip/widgets/store.dart';
+import 'package:zulip/widgets/subscription_list.dart';
+import 'package:zulip/widgets/unread_count_badge.dart';
+
+import '../model/binding.dart';
+import '../example_data.dart' as eg;
+
+void main() {
+  TestZulipBinding.ensureInitialized();
+
+  Future<void> setupStreamListPage(WidgetTester tester, {
+    required List<Subscription> subscriptions,
+    UnreadMessagesSnapshot? unreadMsgs,
+  }) async {
+    addTearDown(testBinding.reset);
+    // create simple versions from subscriptions
+    final List<ZulipStream> streams = subscriptions.map(
+      (e) => eg.stream(streamId: e.streamId, name: e.name)).toList();
```

Comment:
```suggestion
    final List<ZulipStream> streams = subscriptions.toList();
```

right? :grin: 

(And then inline that to the line below that uses it.)